### PR TITLE
Wasm binding update

### LINKED ISSUE #3358: WASM output binding content
cc @codefromthecrypt

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [n/a] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [n/a] Images use HTML style and have alternative text
- [n/a] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This adds missing documentation for the WebAssembly a.k.a. Wasm output binding. Note: this also corrects case format, as in WebAssembly, when capitalized it is Wasm, not WASM.

## Issue reference

this PR will close: #3358

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
link fail because [this](https://github.com/dapr/docs/pull/3356) needs to pick into v1.11, already in v1.10

**Comment by hhunter-ms** [comment]
@codefromthecrypt the link fix was merged into v1.11 yesterday. could you try pulling in v1.11?

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+title: "Wasm"
+linkTitle: "Wasm"
+description: "Detailed documentation on the WebAssembly binding component"
+aliases:
+- "/operations/components/setup-bindings/supported-bindings/wasm/"
+---
+
+## Overview
+
+WebAssembly is a way to safely run code compiled in other languages. Runtimes
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
With WebAssembly, you can safely run code compiled in other languages. Runtimes
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+execute WebAssembly Modules (Wasm), which are most often binaries with a `.wasm`
+extension.
+
+The Wasm Binding allows you to invoke a program compiled to wasm, by passing
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
The Wasm Binding allows you to invoke a program compiled to Wasm by passing
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+commandline args or environment variables to it, similar to how you would with
+a normal subprocess. For example, you can satisfy an invocation using Python
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
a normal subprocess. For example, you can satisfy an invocation using Python,
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+even though Dapr is written in go and running on a platform that doesn't have
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
even though Dapr is written in Go and is running on a platform that doesn't have
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+Python installed!
+
+The Wasm binary must be a program compiled with the WebAssembly System
+Interface (WASI). The binary can be a program you've written such as in Go, or
+an interpreter you use to run inlined scripts, such as Python.
+
+Minimally, a user must specify a Wasm binary compiled with the canonical WASI
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
Minimally, you must specify a Wasm binary compiled with the canonical WASI
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+version `wasi_snapshot_preview1` a.k.a. `wasip1`, often abbreviated to `wasi`.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
version `wasi_snapshot_preview1` (a.k.a. `wasip1`), often abbreviated to `wasi`.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+If compiling yourself in Go 1.21+, this is `GOOS=wasip1 GOARCH=wasm`, whereas
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
> **Note:** If compiling in Go 1.21+, this is `GOOS=wasip1 GOARCH=wasm`. In TinyGo, Rust, and Zig, this is the target `wasm32-wasi`.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+in TinyGo, Rust and Zig, this is the target `wasm32-wasi`.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+You can also re-use an existing binary. For example, [Wasm Language Runtimes](https://github.com/vmware-labs/webassembly-language-runtimes)
+distributes interpreters including PHP, Python and Ruby, already compiled to
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
distributes interpreters (including PHP, Python, and Ruby) already compiled to
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+WASI.
+
+Wasm binaries are loaded from a URL. For example, a URL "file://rewrite.wasm"
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
Wasm binaries are loaded from a URL. For example, the URL `file://rewrite.wasm`
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+loads "rewrite.wasm" from the current directory of the process. On Kubernetes,
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
loads `rewrite.wasm` from the current directory of the process. On Kubernetes,
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+see [mounting volumes to the Dapr sidecar]({{< ref kubernetes-volume-mounts.md >}})
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
see [How to: Mount Pod volumes to the Dapr sidecar]({{< ref kubernetes-volume-mounts.md >}})
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+to configure a filesystem mount that can contain Wasm binaries.
+
+Dapr embeds [wazero](https://wazero.io) which loads programs compiled with the
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
Dapr embeds [wazero](https://wazero.io), which loads programs compiled with the
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+WebAssembly System Interface (WASI), obviating platform dependencies or forking
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
WASI, obviating platform dependencies or forking
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+otherwise needed to accomplish something like this.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]: > Dapr embeds wazero, which loads programs compiled with the WASI, obviating platform dependencies or forking otherwise needed to accomplish something like this.

This sentence is a little odd... is there a way you can reword this to make it more clear?

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+
+## Component format
+
+To configure a WebAssembly (Wasm) binding, create a component of type
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
To configure a Wasm binding, create a component of type
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+  metadata:
+  - name: url
+    value: "file://uppercase.wasm"
+```
+
+## Spec metadata fields
+
+| Field | Details                                                        | Required | Example        |
+|-------|----------------------------------------------------------------|----------|----------------|
+| url   | The URL of the resource including the Wasm binary to instantiate. The supported schemes include "file://". The path of a "file://" URL is relative to the Dapr process unless it begins with '/'. | true     | "file://hello.wasm" |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
| url   | The URL of the resource including the Wasm binary to instantiate. The supported schemes include `file://`. The path of a `file://` URL is relative to the Dapr process unless it begins with `/`. | true     | `file://hello.wasm` |
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+
+## Example request
+
+The `data` field, if present will be the program's STDIN. You can optionally
+pass metadata properties with each request:
+
+- `args` any CLI arguments, comma-separated. This excludes the program name.
+
+For example, if your Wasm binary is `ruby.wasm`, the following request would
+have the response data "Hello, salaboy"
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
have the response data "Hello, salaboy":
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
quick grammar/consistency review

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md (hunk 15 above)
Comment (codefromthecrypt) [reply daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]: sure. I'll give a go on rebase! thanks for all the help so far

Comment (codefromthecrypt) [reply daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]: tried in https://github.com/dapr/docs/pull/3362/commits/8bcf61570e5b1a0bce76cf6835195411c647de53 lemme know

**Comment by codefromthecrypt** [comment]
thanks for the epic help @hhunter-ms!

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md (hunk 15 above)
Comment (hhunter-ms) [reply daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]: Awesome, much clearer! Thanks!

**Review by hhunter-ms — APPROVED** [review]

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+
+## Overview
+
+With WebAssembly, you can safely run code compiled in other languages. Runtimes
+execute WebAssembly Modules (Wasm), which are most often binaries with a `.wasm`
+extension.
+
+The Wasm Binding allows you to invoke a program compiled to Wasm by passing
+commandline args or environment variables to it, similar to how you would with
+a normal subprocess. For example, you can satisfy an invocation using Python,
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md:18]:
```suggestion
a normal subprocess. For example, you can invoke methods in an app written in  Python,
```

**Comment by msfussell** [comment]
@codefromthecrypt - One small suggested change.

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+- `execute`
+
+## Example request
+
+The `data` field, if present will be the program's STDIN. You can optionally
+pass metadata properties with each request:
+
+- `args` any CLI arguments, comma-separated. This excludes the program name.
+
+For example, if your Wasm binary is `ruby.wasm`, the following request would
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]: Do you need to example what is ruby.wasm here, or does this just contain the wasi runtime?

Comment (codefromthecrypt) [reply daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]: sounds like I need a little more context. np

Comment (codefromthecrypt) [reply daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]: tried in https://github.com/dapr/docs/pull/3362/commits/81a53f826bf67da141564c1aa247948370f005d8

**Comment by hhunter-ms** [comment]
@codefromthecrypt one day this build will succeed lol - could you pull in the link fix from v1.11?

**Comment by codefromthecrypt** [comment]
thanks for the help landing this @hhunter-ms and @msfussell!
